lib/context.js

```diff
@@ -238,13 +229,19 @@ class ResponseContext {
             }
           : undefined),
       };
+
       const authParams = {
         ...options.authorizationParams,
         ...authVerification,
       };
 
+      const usePKCE =
+        options.authorizationParams.response_type.includes('code');
       if (usePKCE) {
-        authVerification.code_verifier = transient.generateNonce();
+        debug(
+          'response_type includes code, the authorization request will use PKCE'
+        );
+        authVerification.code_verifier = transient.generateCodeVerifier();
```

Comment:
`generateNonce` and `generateCodeVerifier` are [identical](https://github.com/panva/node-openid-client/blob/main/lib/helpers/generators.js#L10-L11) but I changed to the latter here to avoid any potential confusion reading this code